tests/proxy_lifetime_tests.cpp

```diff
@@ -39,11 +39,17 @@ TEST(ProxyLifetimeTests, TestDefaultConstrction) {
   ASSERT_FALSE(p.has_value());
 }
 
-TEST(ProxyLifetimeTests, TestNullConstrction) {
+TEST(ProxyLifetimeTests, TestNullConstrction_Nullptr) {
   pro::proxy<details::TestFacade> p = nullptr;
   ASSERT_FALSE(p.has_value());
 }
 
+TEST(ProxyLifetimeTests, TestNullConstrction_TypedNullPointer) {
+  int* v = nullptr;
+  pro::proxy<utils::spec::Stringable> p = v;
+  ASSERT_FALSE(p.has_value());
```

Comment:
I'd expect that `p.has_value()` returns `true` because it has a delegated instance although the instance's value is `nullptr`.
Suggest using other way to provide diagnostics when user try to deference an underlying `nullptr` instance.

Reply:
I agree with your point and have closed the linked issue #242. I will abandon this PR and file another one.